src/main/java/com/lothrazar/storagenetwork/capability/handler/MasterItemStackHandler.java

```diff
@@ -0,0 +1,90 @@
+package com.lothrazar.storagenetwork.capability.handler;
+
+import java.util.List;
+import javax.annotation.Nonnull;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.ItemHandlerHelper;
+
+import com.lothrazar.storagenetwork.block.main.TileMain;
+
+/**
+ * ItemStackHandler used for interfacing with TileMain like a chest.
+ */
+public class MasterItemStackHandler extends ItemStackHandlerEx {
+  
+  private TileMain tileMain;
+  
+  public MasterItemStackHandler(TileMain tileMain) {
+    this(tileMain, Integer.MAX_VALUE);
+  }
```

Comment:
I just realized a problem with using Integer.MAX_VALUE. The list inside the ItemStackHandler itself should be non-null, so the entire list will be backed by an array of size Integer.MAX_VALUE.

I could try to make the "size" of the inventory dynamic, if this negatively impacts any testing @Lothrazar can do, but also I don't know if changing the size of this inventory would have negative consequences. Could just use a smaller, big number. Or whatever the actual stack limit in the network currently is.